services/workflows-service/src/transaction/transaction.repository.ts

```diff
@@ -59,22 +58,24 @@ export class TransactionRepository {
   async findManyWithFilters(
     getTransactionsParameters: GetTransactionsDto,
     projectId: string,
+    options?: Prisma.TransactionRecordFindManyArgs,
   ): Promise<TransactionRecord[]> {
     const args: Prisma.TransactionRecordFindManyArgs = {};
     if (getTransactionsParameters.page?.number && getTransactionsParameters.page?.size) {
-      // Temporary fix for pagination (class transfomer issue)
+      // Temporary fix for pagination (class transformer issue)
       const size = parseInt(getTransactionsParameters.page.size as unknown as string, 10);
       const number = parseInt(getTransactionsParameters.page.number as unknown as string, 10);
 
-      args.take = number;
-      args.skip = (size - 1) * number;
+      args.take = size;
+      args.skip = size * (number - 1);
```

Comment:
`size` of 50 and `number` of `1` would return a single transaction instead of 50 before this change.